instrumentation/opentelemetry-instrumentation-falcon/src/opentelemetry/instrumentation/falcon/__init__.py

```diff
@@ -193,7 +211,6 @@ def __call__(self, env, start_response):
         env[_ENVIRON_ACTIVATION_KEY] = activation
 
         def _start_response(status, response_headers, *args, **kwargs):
-            otel_wsgi.add_response_attributes(span, status, response_headers)
```

Comment:
This was a pretty insidious bug that was hard to figure out.

If you look at the middleware, it is already setting `span.status`. This _re sets it_. `otel_wsgi.add_response_attributes` sets the same status code as the middleware, but does _not_ set the description. So after this line executes `span.status.description` is always None. But none of the tests were checking this, so it was not caught.

But what about `test_500` then? Well, in Falcon 2 (but not 3) `super().__call__(...)` would re-raise the error, triggering the `except` clause right below this that skips calling `_start_response` and thus `otel_wsgi.add_response_attributes(span, status, response_headers)` is never called and `span.status.description` is carried over from the middleware.

I'll add an assertion for `span.status.description` to other tests so that this can be avoided in the future.

Reply:
Checks added in 8eda7b6916989b74d03c9d3f506532e8b833e614